crates/re_viewport/src/context_menu.rs

```diff
@@ -0,0 +1,296 @@
+//TODO(ab): use list items to make those context menu nice to look at
+
+use crate::{Contents, ViewportBlueprint};
+use itertools::Itertools;
+use re_log_types::{EntityPath, EntityPathFilter};
+use re_space_view::{DataQueryBlueprint, SpaceViewBlueprint};
+use re_viewer_context::{ContainerId, Item, SpaceViewClassIdentifier, ViewerContext};
+
+/// Trait for things that can populate a context menu
+trait ContextMenuItem {
+    //TODO(ab): should probably return `egui::WidgetText` instead
```

Comment:
so.. why not? :)

Reply:
That comment is actually wrong. The end-game is to return a `ListItem` (with all the fanciness that comes with it: nicer look, icon, etc.). Both `ListItem` and the egui context menu code needs a bit of a overhaul for this to work though, so it'll come as a follow-up.